### ISSUE TITLE
macOS Big Sur DNS failures using exit-nodes (sometimes)

### ISSUE THREAD

**Issue by DentonGentry** [tailscale/tailscale#1647]
[Copying from earlier issue]

On macOS Big Sur, tailscale will set itself as a default route on a tailnet with exit nodes enabled and ACLs set to allow every IP on the internet but the CGNAT range, making the MacBook unable to connect to any external destinations unless you enable an exit node.

Here are the relevant logs of my MacBook in this state:
```
26 Mar 21 11:51 -0400: IPv4 packet with disallowed source address from [XNzWu]
26 Mar 21 11:51 -0400: IPv4 packet with disallowed source address from [XNzWu]
26 Mar 21 11:51 -0400: IPv4 packet with disallowed source address from [XNzWu]
26 Mar 21 11:51 -0400: IPv4 packet with disallowed source address from [XNzWu]
26 Mar 21 11:51 -0400: v1.6.0-tddc975fcb-g995460c32 peers: 568/65096 3060200/663592
```

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2yk0h)

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_2zrf5)

**Comment by DentonGentry** [comment]
If I have Tailscale connected, and an exit node enabled, and DNS servers configured in https://login.tailscale.com/admin/dns, I think I'm seeing DNS resolution failures. I can still reach https://1.1.1.1/ in Chrome, and can ping IP addresses, but reaching websites by name (mostly) fails.

If there are no DNS servers in https://login.tailscale.com/admin/dns I have not produced a problem.

**Comment by DentonGentry** [comment]
With exit node enabled:
- If I make my DNS server the 100.x.x.x address of the DNS server in my house, I don't see a problem. I see DNS queries and responses on utun2.
- If I make my DNS server 8.8.8.8, I can see DNS queries on utun2 to 8.8.8.8:53 but see no responses. Not even errors, just no responses.

**Comment by DentonGentry** [comment]
I do see the default route, always installed whether there is an exit node or not:

```
Internet:
Destination        Gateway            Flags        Netif Expire
default            10.1.10.1          UGSc           en0
default            link#19            UCSI         utun2
```

However I don't see issues arising from this.

The "I" in the flags field means it is of interface scope: packets will only use that route if they have already decided to use the tun2 interface, either because of SO_BINDTODEVICE (which we do) or some other reason.

When Darshini produced the problem in https://github.com/tailscale/corp/issues/1502#issuecomment-812226881 there is additionally a "c" in the flags field, which means `RTF_PRCLONING: Protocol-specified generate new routes on use` I'm not entirely sure what that means.

**Comment by DentonGentry** [comment]
I do see the disallowed source address log message:

```
% ./cmd/logprint/logprint -i 46f5079a9f5b7a38d61dcda551da6bce3b01b69d0778725d574eaf986845c7bc --since 1h | grep disallowed
collection: tailnode.log.tailscale.io, logid: 46f5079a9f5b7a38d61dcda551da6bce3b01b69d0778725d574eaf986845c7bc
2021-04-02 07:27:08.663439 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:28:51.09557 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:29:07.951801 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:29:38.158709 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:30:08.366513 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:30:38.583974 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:31:08.876069 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:31:38.990969 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:32:09.197805 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:32:39.406499 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
2021-04-02 07:32:45.416513 -0700 PDT: IPv4 packet with disallowed source address from [4bDH0]
```

**Comment by DentonGentry** [comment]
The log message comes from wireguard-go:

```
func (peer *Peer) RoutineSequentialReceiver() {
        device := peer.device
        ...
                switch elem.packet[0] >> 4 {
                case ipv4.Version:
                        ...
                        if device.allowedips.LookupIPv4(src) != peer {
                                device.log.Verbosef("IPv4 packet with disallowed source address from %v", peer)
                                goto skip
                        }
```

**Comment by DentonGentry** [comment]
To reproduce:
- set up a machine to be an exit node. I was additionally having it advertise routes of 0.0.0.0/0,::/0 though it isn't clear that is necessary.
- (probably) add DNS servers in admin panel. I did not see a problem until I did so.
- Enable use of the exit node on the Mac

For me it only manifests as DNS failures. I can reach https://1.1.1.1, I can ping nodes by IP address, etc.

Xe and Darshini have also reproduced it and I think it manifests as a more general loss of connectivity.

**Comment by DentonGentry** [comment]
Darshini has reproduced it in a different Tailnet, and we now think the symptoms are the same. DNS, specifically, is not making it through. https://1.1.1.1, which is able to work because CloudFlare encodes their IP addresses in the certificate as alternative names, is reachable.

**Comment by bradfitz** [comment]
On macOS (Big Sur):

* no exit node + "Use Tailscale DNS Settings" on => /etc/resolv.conf is 100.100.100.100, DNS works
* no exit node + "Use Tailscale DNS Settings" off => /etc/resolv.conf is my home one, DNS doesn't work
* exit node + "Use Tailscale DNS Settings" on => /etc/resolv.conf is 100.100.100.100, DNS works
* exit node + "Use Tailscale DNS Settings" off => **/etc/resolv.conf is empty, DNS doesn't work**

**Comment by DentonGentry** [comment]
That's... interesting. Oddly we've also produced the problem where we could see packets to 8.8.8.8:53 on utun2, with no DNS responses coming back. I did not check /etc/resolv.conf during that time, though there must have been some DNS configuration in `scutil --dns` or it wouldn't have sent DNS packets at all.

Trying to produce it again now. to check Maddeningly, it isn't failing now. I've enabled the exit node, yet everything is working.

**Comment by bradfitz** [comment]
I was running HEAD but I went back to 1.6.0 and can still reproduce.  Console.app logs say, when I enable the exit node:

```
66633	default	11:52:02.202490-0700	IPNExtension	SetPrefs: Prefs{ra=true dns=false want=true exit=nYT9HC6CNTRL Persist{lm=, o=[DP1j3], n=[O8p89] u="bradfitz@tailscale.com"}}
66633	default	11:52:02.205698-0700	IPNExtension	wgengine: Reconfig: configuring userspace wireguard config (with 7/83 peers)
66633	default	11:52:02.206566-0700	IPNExtension	wgengine: Reconfig: configuring router
66633	default	11:52:02.206703-0700	IPNExtension	setRoutesFunc called
66633	default	11:52:02.208748-0700	IPNExtension	[Extension io.tailscale.ipn.macos.network-extension]: provider set tunnel configuration to 
    tunnelRemoteAddress = <9-char-str>
    DNSSettings = {
        protocol = cleartext
        server = ()
        searchDomains = ()
        matchDomains = (
            <0-char-str>,
        )
        matchDomainsNoSearch = NO
    }
    IPv4Settings = {
        configMethod = PPP
        addresses = (
            <14-char-str>,
        )
        subnetMasks = (
            255.255.255.255,
        )
        includedRoutes = (
.....
        )
        overridePrimary = NO
    }
    IPv6Settings = {
        configMethod = automatic
        addresses = (
            <39-char-str>,
        )
        networkPrefixLengths = (
            48,
        )
        includedRoutes = (
            {
                destinationAddress = <16-char-str>
                destinationNetworkPrefixLength = 48
            },
            {
                destinationAddress = <2-char-str>
                destinationNetworkPrefixLength = 0
            },
        )
    }
    MTU = 1280
```

**Comment by DentonGentry** [comment]
Debugging session:
- Problem #1: we may be asking NetworkExtension to install us as a default nameserver, even if we don't have any nameserver addresses. Without an exit-node/default route it looks like MacOS ignores this, but if there is a default route it appears to go ahead and write out an empty DNS config. This results in the [symptom noted about an empty /etc/resolv.conf](https://github.com/tailscale/tailscale/issues/1647#issuecomment-812649522)
- Problem #2: still debugging.

**Comment by DentonGentry** [comment]
We asked the user who first reported this issue whether "Use Tailscale DNS Settings" is checked in the debug options on the Tailscale menu in the MacOS Menubar. They confirmed that it is checked.

**Comment by danderson** [comment]
Summary of debugging:
 - One issue we hit was only a thing at HEAD, caused by recent DNS rework. We were installing 100.100.100.100 as a resolver even though the internal forwarder was unconfigured. I pushed a fix for that, but this issue is not in 1.6.0.
 - macOS NetworkExtension config contains both network settings (routes etc.) and DNS settings (resolvers, search domains...). The DNS setting behavior varies based on whether the routing config contains a default route:
   - If the config does _not_ set a default route, an empty DNS configuration (no resolvers etc.) translates to no new resolver added in `scutil --dns`.
   - If the config _does_ set a default route, an empty DNS configuration translates into a "blackhole resolver" being installed that overrides all existing configuration and null-routes DNS.
   - IOW, if you install a default route with NetworkExtension, you _must_ provide a resolver configuration that can resolve all names, because macOS will disable all other resolvers in the system.
   - This behavior is consistent across Catalina and Big Sur. We plan to fix it with the big DNS rework in Tailscale 1.8.
 - There _may_ be another issue at play, where the system default routes and the Tailscale data plane config get out of sync, so we keep sending default routes to Tailscale but then drop them in the WireGuard engine. This is unconfirmed at this time, because a lot of indications we've relied on so far turn out to be red herrings:
   - A default route in `netstat -nr` pointing to the Tailscale utun is _not_ relevant, if that route has the `I` flag set. Those routes are interface-scoped routes, which only take effect with sockets bound to specific interfaces. These routes are not "true" default routes.
   - WireGuard complaints of `IPv4 packet with disallowed source address` are not relevant. When we turn off exit nodes on the system, we reconfigure WireGuard to no longer accept 0.0.0.0/0 and ::/0 from the exit node (i.e. "I no longer trust you to send packets from the internet"). Any in-flight connections going over the exit node will continue to send backscatter over WireGuard for a while (e.g. TCP retransmits, DNS responses that raced with the exit node disable). That backscatter shows up as "disallowed source" log entries in WireGuard.
   - Based on user reports, this issue does _seem_ to be different from the DNS issues above, but our only hypothesis so far is that NetworkExtension is misapplying configuration changes and leaving a "true" default route in place. However, we have no hard evidence for this yet, since the prior two points were our "proof" and it turns out they're irrelevant.

**Comment by clarity99** [comment]
I'm also seeing this problem on bigsur 11.2.2. As soon as I set the exit node, dns stops working. It doesn't matter if I have Use Tailscale DNS settings on/off.

**Comment by DentonGentry** [comment]
One note: exit nodes does (deliberately) block access to the LAN. Its model is that you are out in a public place / guest Wifi / etc where other nodes on the LAN are untrusted. So if your usual DNS server is within your home or office, it would be blocked when an exit node is enabled.

If you configure a DNS server of 8.8.8.8 or 1.1.1.1 or some other public address in https://login.tailscale.com/admin/dns and end up with an empty list in `scutil --dns`, then you're hitting this issue.

**Comment by clarity99** [comment]
i have set my dns to 1.1.1.1 and 8.8.8.8. /etc/resolv.conf is empty and the output from scutil --dns is:
```
❯ scutil --dns
DNS configuration

resolver #1
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1
  nameserver[0] : 1.1.1.1
  nameserver[1] : 8.8.8.8
  if_index : 9 (en0)
  flags    : Scoped, Request A records
  reach    : 0x00000002 (Reachable)
```

**Comment by bradfitz** [comment]
Does toggling MagicDNS make a difference?

**Comment by clarity99** [comment]
It does - if I enable it, it works through the exit node!

**Comment by DentonGentry** [comment]
Current plan is to suppress updating DNS server entries in the Swift code.

**Comment by clarity99** [comment]
Just to add - I noticed that the exit node also wasn't working from iOS and
once I enabled magic dns it started working there as well.

**Comment by DentonGentry** [comment]
I applied what looked like a fix based on what we know of the problem, changing this:

```
netSettings.dnsSettings = dns
```

to this:

```
if dns.servers.count > 0 {
    netSettings.dnsSettings = dns
}
```

I later added some logging to ensure that we were taking the dns.servers.count == 0 path and not setting DNS servers.

The problem remains unchanged. I enable an exit node, and all name resolution stops and /etc/resolv.conf becomes empty.

Looking at it more, I think the root cause is slightly more complex (at least in my setup): the DNS servers added by DHCP on the Wi-Fi interface are always scoped:
```
% scutil --dns
DNS configuration

resolver #1
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records, Request AAAA records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1.                                                    <-- HERE
  search domain[0] : localdomain
  nameserver[0] : 2602:248:7b4a:ce60::1
  nameserver[1] : 10.1.10.1
  if_index : 9 (en0)
  flags    : Scoped, Request A records, Request AAAA records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)
```

Note that resolver 1 at the end of the output is scoped to if_index=9.

So the root cause is not: **Tailscale adds an empty DNS configuration, causing the existing config to disappear**
I think the root cause is: **Tailscale adds a default route which makes the en0 DNS config unreachable**

Packets are no longer allowed to egress en0 once the default route is added.

**Comment by DentonGentry** [comment]
I checked a MacOS 10.15 Catalina system and see the same behavior. The DNS servers learned from DHCP on Wi-Fi are scoped to en0, and are not useable when the exit node is activated.

**Comment by DentonGentry** [comment]
Found a MacOS 10.13 High Sierra system, and its DNS servers are scoped to the Wi-Fi interface as well.

My Big Sur machine is back to being able to produce the problem. If the problem goes away again Imma be really curious how.

**Comment by poswald** [comment]
I've been following this issue as I am also getting this intermittently. I'm on 11.2.3 Big Sur with no exit nodes configured (AFAIK) and I have 1.1.1.1 configured up in the DNS settings screen and Magic DNS enabled.

My home network uses the 10.0.4.x range, and I am currently connected via WiFi:

```
$ scutil --dns
DNS configuration

resolver #1
  nameserver[0] : 10.0.4.1
  if_index : 5 (en0)
  flags    : Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1
  nameserver[0] : 10.0.4.1
  if_index : 5 (en0)
  flags    : Scoped, Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)
```

When I connect tailscale:
```
$ scutil --dns
DNS configuration

resolver #1
  search domain[0] : pauloswald.gmail.com.beta.tailscale.net
  nameserver[0] : 100.100.100.100
  if_index : 17 (utun2)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 100000

resolver #2
  nameserver[0] : 10.0.4.1
  if_index : 5 (en0)
  flags    : Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)
  order    : 200000

resolver #3
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #4
  domain   : pauloswald.gmail.com.beta.tailscale.net
  nameserver[0] : 100.100.100.100
  if_index : 17 (utun2)
  flags    : Supplemental, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
  order    : 100001

resolver #5
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #6
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #7
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #8
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #9
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1
  nameserver[0] : 10.0.4.1
  if_index : 5 (en0)
  flags    : Scoped, Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)

resolver #2
  search domain[0] : pauloswald.gmail.com.beta.tailscale.net
  nameserver[0] : 100.100.100.100
  if_index : 17 (utun2)
  flags    : Scoped, Request A records, Request AAAA records
  reach    : 0x00000003 (Reachable,Transient Connection)
```

Let me know if this information is at all helpful or if there is additional info you would want.

**Comment by danderson** [comment]
The core issue here is that macOS NetworkExtension behaves differently if the tunnel interface is considered "primary" - meaning it has a default route. If that happens, the tunnel interface's DNS settings become primary unconditionally, replacing other DNS settings. This means that when we enable an exit node, we _must_ provide some DNS settings, otherwise the effective DNS configuration will be empty.

This is patched up in 1e5c608faec8c156f5fc4f710722b7f93dc73e0e for 1.8, and will work in most cases: this will force the use of a user-selected fallback DNS resolver when an exit node is active on macOS, ensuring that _some_ configuration still exists.

Meanwhile, #1713 is the fully correct fix for this (always kick DNS over to the exit node for resolution), but won't make the cut for 1.8.

Marking this bug fixed since we have a workaround ready for 1.8, and moving the tracking of the full fix to #1713.

**Comment by morpig** [comment]
hm, apparently this issue is fixed? haven't tried exit node in a long time (since #1544), but have updated to the latest version. still having issues where dns is not resolving.

running macos 11.2.3, tailscale 1.10.1.

not using magic dns/nameservers from tailscale's dashboard.

checked /etc/resolv.conf, it's empty:
```
dito@Ditos-MacBook-Air ansible % cat /etc/resolv.conf 
#
# macOS Notice
#
# This file is not consulted for DNS hostname resolution, address
# resolution, or the DNS query routing mechanism used by most
# processes on this system.
#
# To view the DNS configuration used by this system, use:
#   scutil --dns
#
# SEE ALSO
#   dns-sd(1), scutil(8)
#
# This file is automatically generated.
#
```

```
dito@Ditos-MacBook-Air ansible % scutil --dns
DNS configuration

resolver #1
  nameserver[0] : 172.16.0.5
  nameserver[1] : 172.16.0.10
  if_index : 11 (en0)
  flags    : Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)

resolver #2
  domain   : local
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300000

resolver #3
  domain   : 254.169.in-addr.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300200

resolver #4
  domain   : 8.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300400

resolver #5
  domain   : 9.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300600

resolver #6
  domain   : a.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 300800

resolver #7
  domain   : b.e.f.ip6.arpa
  options  : mdns
  timeout  : 5
  flags    : Request A records
  reach    : 0x00000000 (Not Reachable)
  order    : 301000

DNS configuration (for scoped queries)

resolver #1
  nameserver[0] : 172.16.0.5
  nameserver[1] : 172.16.0.10
  if_index : 11 (en0)
  flags    : Scoped, Request A records
  reach    : 0x00020002 (Reachable,Directly Reachable Address)
```

Any help?
